### PR TITLE
Added custom Cairo paths.

### PR DESCRIPTION
**Pull request by Radinyn**
#29

PS:
You might consider using a formatter on this repository to decrease diff sizes.

### CODE REVIEW

**Comment by agolajko** [comment]
Keen to have this working, getting cairo-ls working with the cairo version in protostar

File: server/src/server.ts [hunk 1]
```diff
@@ -791,17 +792,27 @@ function getCompileCommand(settings: CairoLSSettings, tempFiles: TempFiles, text
...
+	const cairoPath = settings.cairoPath;
 
-	if (workspaceFolders.length > 0) {
+	if (workspaceFolders.length > 0 || cairoPath.length > 0) {
 		cairoPathParam = '--cairo_path=';
-		for (i = 0; i < workspaceFolders.length; i++) {
-			cairoPathParam += appendSourceDir(workspaceFolders[i], sourceDir);
+
+		for (i = 0; i < cairoPath.length; i++) {
+			cairoPathParam += cairoPath[i]
```
Comment (ericglau) [inline server/src/server.ts:801]:
```suggestion
			cairoPathParam += cairoPath[i];
```

File: server/src/server.ts [hunk 2]
```diff
@@ -791,17 +792,27 @@ function getCompileCommand(settings: CairoLSSettings, tempFiles: TempFiles, text
...
+
 			if (i < workspaceFolders.length - 1) {
```
Comment (ericglau) [inline server/src/server.ts:803]: This should be checking `cairoPath.length` instead of `workspaceFolders.length`

File: server/src/server.ts [hunk 3]
```diff
@@ -791,17 +792,27 @@ function getCompileCommand(settings: CairoLSSettings, tempFiles: TempFiles, text
...
 				cairoPathParam += ':';
 			}
 		}
+
+		for (i = 0; i < workspaceFolders.length; i++) {
+			cairoPathParam += ':';
```
Comment (ericglau) [inline server/src/server.ts:809]: If `cairoPath` was empty, this appends `:` at the beginning which isn't correct.

File: server/src/server.ts [hunk 4]
```diff
@@ -1392,8 +1403,15 @@ async function initPackageSearchPaths(uri: string) {
 
 		const settings = await getDocumentSettings(uri);
 		const sourceDir = settings.sourceDir;
+		const cairoPath = settings.cairoPath;
 
 		packageSearchPaths = '';
+
+		for (let i = 0; i < cairoPath.length; i++) {
+			packageSearchPaths += cairoPath[i];
```
Comment (ericglau) [inline server/src/server.ts:1411]: Looks like this should append `;` after each path as well.

**Review by ericglau — COMMENTED** [review]
Thanks for the PR!  Just a few comments.

File: server/src/server.ts (hunk 1 above)
Comment (Radinyn) [reply server/src/server.ts]: done

File: server/src/server.ts (hunk 2 above)
Comment (Radinyn) [reply server/src/server.ts]: done

File: server/src/server.ts (hunk 3 above)
Comment (Radinyn) [reply server/src/server.ts]: done

File: server/src/server.ts (hunk 4 above)
Comment (Radinyn) [reply server/src/server.ts]: done

**Review by ericglau — APPROVED** [review]
LGTM, thanks!

**Pull request merged**